src/common/commonutils/FileUtils.c

```diff
@@ -423,5 +423,43 @@ int CheckFileSystemMountingOption(const char* mountFileName, const char* mountDi
         OsConfigLogError(log, "CheckFileSystemMountingOption: could not open file '%s', setmntent() failed (%d)", mountFileName, status);
     }
 
+    return status;
+}
+
+int CheckPackageInstalled(const char* packageName, void* log)
+{
+    const char* commandTemplate = "dpkg -l %s | grep ^ii";
+    char* command = NULL;
+    size_t packageNameLength = 0;
+    int status = ENOENT;
+
+    if ((NULL == commandTemplate) || (NULL == packageName) || ((0 == (packageNameLength = strlen(packageName)))))
+    {
+        OsConfigLogError(log, "CheckPackageInstalled called with invalid argument");
+        return EINVAL;
+    }
+    
+    packageNameLength += strlen(commandTemplate) + 1;
+
+    if (NULL == (command = (char*)malloc(packageNameLength)))
+    {
+        OsConfigLogError(log, "CheckPackageInstalled: out of memory");
+        return ENOMEM;
+    }
+
+    memset(command, 0, packageNameLength);
+    snprintf(command, packageNameLength, commandTemplate, packageName);
+
+    if (0 == (status = ExecuteCommand(NULL, command, false, false, 0, 0, NULL, NULL, log)))

```

Comment:
>ExecuteCommand [](http://example.com/codeflow?start=23&length=14)

Should this handle the case where dpkg is not available? For example, Mariner?

Reply:
There is no better option that I am aware of. If you know of one more universal, suggest. Btw, apt (more high level) uses dpkg.